mycroft/skills/skill_loader.py

```diff
@@ -0,0 +1,247 @@
+# Copyright 2019 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Periodically run by skill manager to load skills into memory."""
+import gc
+import imp
+import os
+import sys
+from time import time
+
+from mycroft.configuration import Configuration
+from mycroft.messagebus import Message
+from mycroft.util.log import LOG
+
+SKILL_MAIN_MODULE = '__init__.py'
+
+
+def _get_last_modified_time(path):
+    """Get the last modified date of the most recently updated file in a path.
+
+    Exclude compiled python files, hidden directories and the settings.json
+    file.
+
+    Arguments:
+        path: skill directory to check
+
+    Returns:
+        int: time of last change
+    """
+    all_files = []
+    for root_dir, dirs, files in os.walk(path):
+        dirs[:] = [d for d in dirs if not d.startswith('.')]
+        for f in files:
+            ignore_file = (
+                    f.endswith('.pyc') or
+                    f == 'settings.json' or
+                    f.startswith('.') or
+                    f.endswith('.qmlc')
+            )
+            if not ignore_file:
+                all_files.append(os.path.join(root_dir, f))
+
+    # check files of interest in the skill root directory
+    return max(os.path.getmtime(f) for f in all_files)
+
+
+class SkillLoader:
+    def __init__(self, bus, skill_directory):
+        self.bus = bus
+        self.skill_directory = skill_directory
+        self.skill_id = os.path.basename(skill_directory)
+        self.load_attempted = False
+        self.loaded = False
+        self.last_modified = 0
+        self.last_loaded = 0
+        self.instance = None
+        self.active = True
+
+    @property
+    def config(self):
+        return Configuration.get()
+
+    @property
+    def is_blacklisted(self):
+        """Boolean value representing whether or not a skill is blacklisted."""
+        blacklist = self.config['skills'].get('blacklisted_skills', [])
+        if self.skill_id in blacklist:
+            return True
+        else:
+            return False
+
+    def reload(self):
+        """Load an unloaded skill or reload unloaded/changed needs if necessary.
```

Comment:
This seems to be 1 character over the 79 character limit.

Reply:
fixed